Backport of

commit 8491c8a0bcbb89c03875f128ed29baf9c05c4e79
Author:     Andre Lorbach <alorbach@adiscon.com>
AuthorDate: Mon Jun 18 09:42:26 2018 +0200
Commit:     Andre Lorbach <alorbach@adiscon.com>
CommitDate: Mon Jun 18 09:54:37 2018 +0200

    testbench: added new ssl test certs (100+ years valid)

diff --git a/tests/tls-basic-vg.sh b/tests/tls-basic-vg.sh
index 2e005bc..e73a900 100755
--- a/tests/tls-basic-vg.sh
+++ b/tests/tls-basic-vg.sh
@@ -12,13 +12,13 @@ fi
 
 TESTPORT=20514
 echo 'Start Receiver...'
-valgrind ./receive -p $TESTPORT -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog" > librelp.out.log &
+valgrind ./receive -p $TESTPORT -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog-client" > librelp.out.log &
 PID=$!
 
 sleep 1
 
 echo 'Send Message...'
-valgrind ./send -t 127.0.0.1 -p $TESTPORT -m "testmessage" -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog"
+valgrind ./send -t 127.0.0.1 -p $TESTPORT -m "testmessage" -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog-client"
 
 echo 'Stop Receiver...'
 kill $PID
diff --git a/tests/tls-basic.sh b/tests/tls-basic.sh
index 8b53b1e..b4128c3 100755
--- a/tests/tls-basic.sh
+++ b/tests/tls-basic.sh
@@ -1,9 +1,9 @@
 #!/bin/bash
 . ${srcdir}/test-framework.sh
-startup_receiver -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog"
+startup_receiver -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog-client"
 
 echo 'Send Message...'
-./send -t 127.0.0.1 -p $TESTPORT -m "testmessage" -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog"
+./send -t 127.0.0.1 -p $TESTPORT -m "testmessage" -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "rsyslog-client"
 
 stop_receiver
 check_output "testmessage"
diff --git a/tests/tls-certs/ca.pem b/tests/tls-certs/ca.pem
index 6324c7d..99925c4 100644
--- a/tests/tls-certs/ca.pem
+++ b/tests/tls-certs/ca.pem
@@ -1,15 +1,29 @@
 -----BEGIN CERTIFICATE-----
-MIICYjCCAc2gAwIBAgIBATALBgkqhkiG9w0BAQUwWDELMAkGA1UEBhMCREUxHTAb
-BgNVBAoTFHJzeXNsb2cgdGVzdCByb290IENBMQswCQYDVQQLEwJDQTEdMBsGA1UE
-AxMUcnN5c2xvZy10ZXN0LXJvb3QtY2EwHhcNMDgwNTIwMTI1ODEyWhcNMTgwNTE4
-MTI1ODI0WjBYMQswCQYDVQQGEwJERTEdMBsGA1UEChMUcnN5c2xvZyB0ZXN0IHJv
-b3QgQ0ExCzAJBgNVBAsTAkNBMR0wGwYDVQQDExRyc3lzbG9nLXRlc3Qtcm9vdC1j
-YTCBnDALBgkqhkiG9w0BAQEDgYwAMIGIAoGAw2s+V+WCK7jx9MLpDD4pO8SCqq6Q
-nK/BptvKM+YeBrV9ud3lq6YgbpNmv3/wig43rqpolqk7PdDxTW/mdXPmM72oKr/N
-Fc2cAyOEXK8JTWiqwc//V4qMAnKFfLOxr1dr7WRD0k4Tc8+BWJMQjL2zmGXiSGEF
-YWYIFHLmnX4ZgyMCAwEAAaNDMEEwDwYDVR0TAQH/BAUwAwEB/zAPBgNVHQ8BAf8E
-BQMDBwYAMB0GA1UdDgQWBBQzYQQgUm0YLNdarJnc2c1LxYVClDALBgkqhkiG9w0B
-AQUDgYEAuGWtH7Jkpa0n/izqQ5ddDQP/LT6taivCwlpEYEU9aumpQPWWxtYywKaP
-RfM1JTMLAiYd8MS7TJ8TYRvvR32Y02Y+OhXn11xERkWvBT2M9yzqX6hDfRueN7RT
-fPWsfm/NBTVojzjaECcTFenZid7PC5JiFbcU6PSUMZ49/JPhxAo=
+MIIE9jCCA16gAwIBAgIIWxfM+RyuD1EwDQYJKoZIhvcNAQELBQAwgYsxEzARBgNV
+BAMTCnJzeXNsb2cgY2ExEDAOBgNVBAsTB0FkaXNjb24xFTATBgNVBAoTDEFkaXNj
+b24gR21iSDEYMBYGA1UEBxMPR3Jvc3NyaW5kZXJmZWxkMQswCQYDVQQIEwJCVzEL
+MAkGA1UEBhMCREUxFzAVBgoJkiaJk/IsZAEZFgdyc3lzbG9nMCAXDTE4MDYwNjEy
+MDA1OFoYDzIxMTgwNTEzMTIwMTA1WjCBizETMBEGA1UEAxMKcnN5c2xvZyBjYTEQ
+MA4GA1UECxMHQWRpc2NvbjEVMBMGA1UEChMMQWRpc2NvbiBHbWJIMRgwFgYDVQQH
+Ew9Hcm9zc3JpbmRlcmZlbGQxCzAJBgNVBAgTAkJXMQswCQYDVQQGEwJERTEXMBUG
+CgmSJomT8ixkARkWB3JzeXNsb2cwggGiMA0GCSqGSIb3DQEBAQUAA4IBjwAwggGK
+AoIBgQDHA1CeNI30en7OKGJ/pWyNSH9rnw8z6egDxdTjtOIichA2/LVPaGZYecI0
+GH/UtkRgM69kFuo6dRaWvr0yp+XPfGlAcFcPg6cYKEoYyb18HsTReyuAcdBcYQYg
+2ZdmAqtqeHAqcb7wSEB8XuXLdD2Flsco3GaT3zFNv7t7QFDW1b2tT/92w+QGtI1T
+r5LCMVbISRGSQzhCv29l3UtWt4SKA9W8yhg5zMJkMExaUJVtViRFlAFzKrUVKFqZ
+8sszy2yXIcTLskJJc5if8o5cpWCrbcXSb5TsHA+4FHOB6ou0z7/s3LwYjof056Zy
+ixj0LM4sTYIEttDuLERRvYTIhtLhlA2BWfAfuW9bGuog0C9xGpt99vniHJlKmMms
+qz5hEDLYziUEfj5qzSxKMcA2VCeO1DgajSedh13VN/sWUyk5aKA71i40Cnp4oK/u
+zeYW7vWl/ttCEFc6x8pzeMdoBjv1Jd9hC0Z2UgttC/KnMTKX3y6NR1PVFjDQ3MIA
+7Bhxj50CAwEAAaNaMFgwDwYDVR0TAQH/BAUwAwEB/zAVBgNVHREEDjAMggpyc3lz
+bG9nIGNhMA8GA1UdDwEB/wQFAwMHBAAwHQYDVR0OBBYEFAwRNp2wnAVEZKyXJkDA
+KFMjnY6aMA0GCSqGSIb3DQEBCwUAA4IBgQCOv2kyg8vZ3jncZuZ4i2k0QbXztZoO
+jMDNqFX2935lK6WrVZ6u1InZzsaqrXXOJions3EvmmAqrdTBW6dyw6V2/lic1gGz
+MhlafAPe0DyQCaXp9iFCH91Hzo94YhuPbne1qdga9jrVTiQIHdIKqVtbiUv7i7mN
+43GssOm4a1guf+Qs5rkuHG4YwiJZhjzhmixfXCerHXykJkpSvBUb6EeKA+p5/w+I
+Wjm9sAkJgqcvrNvOwwBZInU1I56zmM0ZwWucSydf0hgOImpgO5F6KGVQRoZsO9IA
+Iju2RQ1y9qVTNp8evVjIMuqXh5ZtU9ti/buZrjr5Zb601jFLZCMrpDVUcVyHUfOB
+rb3nkUFDcFcrKlfovwp/fvlISKM0bp6pFas5X0FXg3sVVI+iIokmHsmqRUHhjdLX
+t27+/TYpyEkjF1cH3acd7MOcw33KxE/4+qUHT2QU9COPkGu7oPS24qDYnmGPLkev
+ZSmpz4jLTmbCBSRcoOB0Q4K36WnegzVtvd8=
 -----END CERTIFICATE-----
diff --git a/tests/tls-certs/cert.pem b/tests/tls-certs/cert.pem
index 6b5b13c..58cdec3 100644
--- a/tests/tls-certs/cert.pem
+++ b/tests/tls-certs/cert.pem
@@ -1,16 +1,31 @@
 -----BEGIN CERTIFICATE-----
-MIIChjCCAfGgAwIBAgIBADALBgkqhkiG9w0BAQUwWDELMAkGA1UEBhMCREUxHTAb
-BgNVBAoTFHJzeXNsb2cgdGVzdCByb290IENBMQswCQYDVQQLEwJDQTEdMBsGA1UE
-AxMUcnN5c2xvZy10ZXN0LXJvb3QtY2EwHhcNMDgwNTIwMTMwNDE5WhcNMTgwNTE4
-MTMwNDI2WjA6MQswCQYDVQQGEwJERTEQMA4GA1UEChMHcnN5c2xvZzEZMBcGA1UE
-CxMQdGVzdCBjZXJ0aWZpY2F0ZTCBnDALBgkqhkiG9w0BAQEDgYwAMIGIAoGAxmHe
-fztJgaGxFYEceiUg0hdMlRVWBqoZelJ8BeXTDnXcu/5F2HtM+l+QDyDaGjKlx+NI
-K4rkj7d6Wd3AKPgOYS0VSDZe3a1xf9rRYzOthWTv7tYi4/LTqPXqN5lKE71dgrB/
-/gOmvV/1YD776FIxVGCSAT0hHwkFC3slmpJSwD8CAwEAAaOBhDCBgTAMBgNVHRMB
-Af8EAjAAMB0GA1UdJQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDATASBgNVHREECzAJ
-ggdyc3lzbG9nMB0GA1UdDgQWBBQYu6eC9UALvC+5K5VOnFRi5OC98TAfBgNVHSME
-GDAWgBQzYQQgUm0YLNdarJnc2c1LxYVClDALBgkqhkiG9w0BAQUDgYEAXaymqsG9
-PNBhhWIRFvXCDMaDM71vUtgSFoNUbxIV607ua2HQosPPM4EHIda6N6hdBK1bMQoG
-yqBwhvw0JVaVaO70Kbs2m2Ypk3YcpJtRqyp8q8+2y/w1Mk1QazFZC29aYgX2iNVf
-X4/x38YEL7Gu5vqPrTn++agnV4ZXECKuvLQ=
+MIIFVzCCA7+gAwIBAgIIWxfNMhQmuncwDQYJKoZIhvcNAQELBQAwgYsxEzARBgNV
+BAMTCnJzeXNsb2cgY2ExEDAOBgNVBAsTB0FkaXNjb24xFTATBgNVBAoTDEFkaXNj
+b24gR21iSDEYMBYGA1UEBxMPR3Jvc3NyaW5kZXJmZWxkMQswCQYDVQQIEwJCVzEL
+MAkGA1UEBhMCREUxFzAVBgoJkiaJk/IsZAEZFgdyc3lzbG9nMCAXDTE4MDYwNjEy
+MDE1NVoYDzIxMTgwNTEzMTIwMjAxWjCBmDEXMBUGA1UEAxMOcnN5c2xvZy1jbGll
+bnQxFTATBgNVBAsTDEFkaXNjb24gR21iSDEVMBMGA1UEChMMQWRpc2NvbiBHbWJI
+MRgwFgYDVQQHEw9Hcm9zc3JpbmRlcmZlbGQxCzAJBgNVBAgTAkJXMQswCQYDVQQG
+EwJERTEbMBkGCgmSJomT8ixkARkWC3JzeXNsb2cuY29tMIIBojANBgkqhkiG9w0B
+AQEFAAOCAY8AMIIBigKCAYEA8nNWVoZwi7fYKSYxm+lidTbMQEHh9pLXV8P1N2FD
+wqhNc6Z71VqQw6gIDyzxBjZf3ldyPb6xAcxikvk2XCxVGqDaP7E1rfnCDSqxcgLG
+r1/TlRH3fFMFPqzMgs0GbDoKi7WuhcNMH1yIRk4uPVSUY3IclcE9sNvMpNVY77c9
+tYeLyoLgG8A2ljlSjbHXDmR8E1C+WcOvjFKQunpv29zIOvfp3Wuw0g3C6E26RyvY
+OnNNq6bmDHXdyIkxYNPwxyfBDGq/WhnrFqQTyEqulVLVVIpK+3fIdtmH/4OpOwxz
+KM0q2k9CzL+AI1JqlFoePeIKYQYAOssUBwB0VBDbhyo8f7txBs6OB1phPcmbkAo2
+bJfL01UT315omqlGWjvvDRjDEeRAfXnWu/UVLIs1MNn09l2+fKwgZyOZIQEk5gGg
+b+xO7pZp5W1IuLxB1DMatQjYK1L1MG8KUebovTyumAj8kOaCJaqa+EWcPvsH2wJb
+zp7JZV7xAMnNQGLHHccY9Iv9AgMBAAGjga0wgaowDAYDVR0TAQH/BAIwADAnBgNV
+HSUEIDAeBggrBgEFBQcDAgYIKwYBBQUHAwEGCCsGAQUFBwMRMCAGA1UdEQQZMBeC
+FXRlc3RiZW5jaC5yc3lzbG9nLmNvbTAPBgNVHQ8BAf8EBQMDB6AAMB0GA1UdDgQW
+BBTFQ9rF9eWMeqNtrDMOGb1fZ54EYTAfBgNVHSMEGDAWgBQMETadsJwFRGSslyZA
+wChTI52OmjANBgkqhkiG9w0BAQsFAAOCAYEAG9hahjpm74U85pP7/zngEJ5P5nnM
+3aC8u4lDM7kodtnxJWoisbv/EBr6SihpDtfWA6+bAJGGu+4UBUaMyUg8w/FjylHc
+INtljbalRuu7YguL2uLdlDNCxQa1XkWugYMObR6r6OvTPUwK0QakTsZuibRbWBDz
+lWKOoM7ZSxZi9DaeChe3yNSGxJSC9xHZbYYBOP5RjChXTZTKtYZemF2wx6EtWRtc
+vXK3TbnJtDy7SPm60EoFnLXUyywmo5mCUSlx85HinS1DzYo7I1yksWQgdW4XgWId
+9RgwPtP0iZSjrZ1TwKqcvo5jvn96LNCoob+JyhpL+9mFvPc6C3vLOxyG7JZOb7Dn
+IpJf52KeBnLRUoIpp+x1XYptoy1ti7r8YqPYn5EHmPxCDVoujskRSX7ncP3SV1qY
+eEgr94Tw/l8GyiaGyixVl2pAMbzYJtgrzQ9UvZurhSPjApGlm87X5KHHXN/CJTY5
+8t59+qvjESRJk74JNpm1L6X/N7HpAsqcsYjP
 -----END CERTIFICATE-----
diff --git a/tests/tls-certs/key.pem b/tests/tls-certs/key.pem
index 3ff507f..4d63250 100644
--- a/tests/tls-certs/key.pem
+++ b/tests/tls-certs/key.pem
@@ -1,15 +1,190 @@
+Public Key Info:
+	Public Key Algorithm: RSA
+	Key Security Level: High (3072 bits)
+
+modulus:
+	00:f2:73:56:56:86:70:8b:b7:d8:29:26:31:9b:e9:62
+	75:36:cc:40:41:e1:f6:92:d7:57:c3:f5:37:61:43:c2
+	a8:4d:73:a6:7b:d5:5a:90:c3:a8:08:0f:2c:f1:06:36
+	5f:de:57:72:3d:be:b1:01:cc:62:92:f9:36:5c:2c:55
+	1a:a0:da:3f:b1:35:ad:f9:c2:0d:2a:b1:72:02:c6:af
+	5f:d3:95:11:f7:7c:53:05:3e:ac:cc:82:cd:06:6c:3a
+	0a:8b:b5:ae:85:c3:4c:1f:5c:88:46:4e:2e:3d:54:94
+	63:72:1c:95:c1:3d:b0:db:cc:a4:d5:58:ef:b7:3d:b5
+	87:8b:ca:82:e0:1b:c0:36:96:39:52:8d:b1:d7:0e:64
+	7c:13:50:be:59:c3:af:8c:52:90:ba:7a:6f:db:dc:c8
+	3a:f7:e9:dd:6b:b0:d2:0d:c2:e8:4d:ba:47:2b:d8:3a
+	73:4d:ab:a6:e6:0c:75:dd:c8:89:31:60:d3:f0:c7:27
+	c1:0c:6a:bf:5a:19:eb:16:a4:13:c8:4a:ae:95:52:d5
+	54:8a:4a:fb:77:c8:76:d9:87:ff:83:a9:3b:0c:73:28
+	cd:2a:da:4f:42:cc:bf:80:23:52:6a:94:5a:1e:3d:e2
+	0a:61:06:00:3a:cb:14:07:00:74:54:10:db:87:2a:3c
+	7f:bb:71:06:ce:8e:07:5a:61:3d:c9:9b:90:0a:36:6c
+	97:cb:d3:55:13:df:5e:68:9a:a9:46:5a:3b:ef:0d:18
+	c3:11:e4:40:7d:79:d6:bb:f5:15:2c:8b:35:30:d9:f4
+	f6:5d:be:7c:ac:20:67:23:99:21:01:24:e6:01:a0:6f
+	ec:4e:ee:96:69:e5:6d:48:b8:bc:41:d4:33:1a:b5:08
+	d8:2b:52:f5:30:6f:0a:51:e6:e8:bd:3c:ae:98:08:fc
+	90:e6:82:25:aa:9a:f8:45:9c:3e:fb:07:db:02:5b:ce
+	9e:c9:65:5e:f1:00:c9:cd:40:62:c7:1d:c7:18:f4:8b
+	fd:
+
+public exponent:
+	01:00:01:
+
+private exponent:
+	63:aa:b0:3a:5f:87:d2:97:df:df:82:98:f0:2b:ef:d2
+	99:95:3b:71:19:ef:e0:0b:70:43:d9:01:4c:15:ab:e6
+	bd:f8:25:df:50:24:ab:d0:12:ba:45:6a:c7:26:33:a2
+	13:67:c5:49:c6:a3:43:46:fe:da:c6:61:e7:60:3e:17
+	f5:39:e2:e6:f9:57:2c:8d:ee:15:b7:de:53:6f:d3:d6
+	43:c7:4b:1c:7e:da:2e:14:c6:ba:47:a6:fa:d5:f0:c2
+	37:0e:97:d0:db:28:e6:d7:86:19:c8:b7:cf:74:7e:8f
+	a6:f0:43:4b:ae:0d:8f:3d:c0:21:08:2f:7b:17:7e:11
+	5b:91:03:5e:17:6a:5b:c6:43:88:41:41:73:53:b2:51
+	23:2b:95:24:97:82:09:e0:43:28:b5:7e:58:56:99:65
+	4c:56:b2:34:3b:fb:0d:b7:2b:02:10:ea:83:06:9f:78
+	98:dc:50:7e:cf:6e:80:9e:98:6f:59:0b:a7:36:67:8b
+	72:ad:75:96:9c:13:a4:8e:17:07:6d:45:30:b1:55:03
+	81:65:6b:02:e1:30:fe:9c:6b:63:80:ae:9d:47:b1:33
+	ea:b6:1c:6d:14:2f:ff:29:71:e2:9c:dd:40:62:34:04
+	44:3d:8d:71:6b:d1:ea:8d:46:63:37:12:26:8f:94:1c
+	f5:6a:d1:66:7f:a9:d6:dd:59:27:79:ce:ea:ad:b3:47
+	62:b9:f6:1c:36:0a:c0:fe:f3:f6:e1:c2:86:12:97:f9
+	f2:7e:62:b4:33:3e:ad:43:05:95:1d:14:fe:be:76:c6
+	44:38:80:a4:25:08:5b:5c:76:4e:a9:bd:aa:7f:f1:05
+	8b:49:81:a0:40:4b:1a:98:d1:c9:cf:98:e1:ba:71:12
+	fc:1f:53:aa:57:5c:65:3a:e0:72:24:d5:8f:27:21:9c
+	fc:b6:d6:f7:c8:07:5c:c2:42:5d:cb:ba:bf:0d:f2:13
+	9b:07:6b:73:ae:1b:f0:b7:1a:6a:fd:20:cf:b2:72:01
+	
+
+prime1:
+	00:fe:14:cf:31:a0:d8:e0:f5:49:35:22:2c:14:13:aa
+	55:d6:26:92:c2:fa:63:02:72:dd:d1:05:d1:0f:b0:a4
+	ae:e4:0e:99:a0:35:e3:84:24:06:aa:27:4d:ce:6f:33
+	6d:8a:96:3e:54:bd:51:92:e9:3f:9b:f2:6e:9d:7d:1e
+	2f:e1:11:bc:db:34:c8:2f:ab:07:ee:4a:5d:96:05:54
+	1c:48:f0:3c:83:99:3a:9f:5d:fe:31:48:0c:b3:0c:ab
+	e3:5c:2d:e7:06:43:91:5c:62:21:76:0a:4f:00:ec:a3
+	95:85:b0:fe:3b:61:8b:07:c7:21:bb:b4:ef:0d:d1:67
+	01:dd:bd:9b:e8:56:eb:ca:22:83:c1:e8:89:3b:fa:17
+	23:e9:de:fd:62:04:73:1f:2a:89:66:d7:d1:16:99:7f
+	98:a0:e0:c2:9b:11:22:23:b9:55:cb:a0:4b:eb:81:a4
+	a8:82:5d:e0:0a:cb:19:fd:18:5d:e2:17:cc:13:2b:19
+	81:
+
+prime2:
+	00:f4:48:0b:16:38:e8:d9:94:7e:38:29:71:d4:2a:fa
+	7c:cd:d8:b5:d2:db:d1:a6:3f:1f:3f:39:2f:f2:af:3d
+	ae:8c:c9:fd:01:32:83:cf:48:f4:f5:25:95:3f:0a:83
+	90:21:75:60:52:01:9a:5e:95:d2:ed:4c:6e:8c:f9:93
+	9a:40:24:b5:91:eb:e6:b4:96:6a:6a:51:1b:9b:18:fd
+	24:c8:c3:68:52:e3:55:f5:23:e0:97:4d:4b:4b:cd:83
+	04:b9:e1:70:b6:4c:42:a6:a0:0c:2e:0d:f0:a1:c0:04
+	08:57:0f:da:0e:0e:ee:32:7c:20:e6:03:ee:35:41:22
+	01:b0:f3:6e:ee:0d:d7:91:1d:ed:98:d4:ef:e6:03:f1
+	39:96:dd:a0:1f:6c:3a:3d:d9:d2:ac:f0:13:73:31:cf
+	b1:fb:13:d3:55:55:1b:6c:bd:39:49:e0:7a:b7:ef:ec
+	3d:5c:dd:7d:bf:a0:11:69:8f:66:fc:45:29:b3:05:18
+	7d:
+
+coefficient:
+	00:9b:0b:66:d7:fe:dd:69:d2:9a:5b:d5:4e:36:f0:c8
+	67:4e:68:ff:d5:21:a1:92:50:27:97:47:6e:82:3d:b3
+	c5:87:d2:c0:15:09:c1:9e:dc:2d:74:5a:cc:2b:ca:82
+	0a:5d:3b:82:93:31:ce:12:bd:5b:7d:54:87:c0:8a:0e
+	df:86:d6:6f:62:ca:4a:41:5f:6c:8e:cb:ac:6e:37:c0
+	b8:26:20:c3:59:37:ce:1f:e7:eb:a5:08:8a:b0:96:f3
+	e0:3d:fc:66:c4:6b:66:14:91:76:2c:22:e4:6a:56:20
+	41:51:6a:77:f1:bf:4c:5b:e8:fa:af:41:09:a8:77:5a
+	6a:66:6d:12:83:54:c1:e8:ff:b7:3a:db:9f:46:e2:ce
+	d8:60:a8:03:23:78:41:5f:fb:ea:16:62:20:24:8d:8b
+	49:64:e4:cb:ca:83:e8:c4:60:97:fb:bf:e7:dc:2c:10
+	ba:03:67:76:ea:4d:00:6c:06:f8:d2:83:fb:26:fe:dc
+	e1:
+
+exp1:
+	00:db:9c:94:f1:ea:7b:4b:05:f8:3e:ea:c8:7a:99:fc
+	d2:b0:85:2a:b9:48:be:a7:71:2d:74:eb:b8:25:fd:a2
+	7a:60:38:96:64:b9:ef:6b:59:d0:40:e6:34:04:48:6d
+	85:33:12:65:d1:cc:c9:63:58:24:58:e0:94:e7:9f:2b
+	a6:6a:bd:70:25:70:5b:cf:87:df:15:02:46:1d:d3:81
+	3d:e9:75:6c:78:ba:ca:c6:d0:f1:53:1b:59:8e:b1:57
+	28:ef:95:e5:67:ae:a3:47:dc:b1:db:5b:51:aa:14:df
+	d7:ef:74:6a:6a:5d:07:54:b3:b3:de:90:a7:47:a4:45
+	8a:81:64:d0:0b:64:85:ed:73:23:bb:49:08:c5:44:87
+	17:cd:9f:a6:17:40:8d:58:d4:e5:6a:75:04:0e:aa:a2
+	75:04:29:01:4c:12:e5:15:47:ae:76:90:98:ce:78:82
+	1f:fc:7e:b3:95:e0:b3:17:e6:25:cf:b4:8d:f9:cb:79
+	01:
+
+exp2:
+	00:ad:d1:a9:99:e9:6a:be:8f:ab:f4:78:81:17:90:71
+	66:66:ee:a2:0a:65:01:db:c2:a4:90:4f:37:ef:fe:83
+	ec:75:1f:b1:18:ae:02:1e:77:f2:5d:91:9d:8e:1b:04
+	1c:85:c2:48:84:71:c1:8f:36:1a:bd:61:9a:74:53:8e
+	94:be:b9:a0:3d:31:d7:9a:7e:41:40:5f:70:3e:3f:5c
+	b9:54:e5:be:90:e4:54:a0:b7:23:b3:65:7b:83:77:bb
+	97:9c:50:6e:ec:38:7d:60:a5:17:a5:99:a3:b4:e4:b5
+	b9:dd:81:59:0e:91:cf:2a:3b:2b:09:a0:98:5a:6a:7b
+	31:2a:e1:3c:f8:c4:fe:b3:b2:d1:02:e3:d9:c4:18:e0
+	07:b3:73:05:e1:f6:17:c7:34:fc:74:b3:eb:93:17:49
+	8b:82:75:cb:f5:b7:36:69:7c:87:8c:2a:e0:07:8b:ff
+	d4:76:04:e0:da:7e:a2:85:33:55:fb:81:97:37:5c:13
+	dd:
+
+
+Public Key ID: C5:43:DA:C5:F5:E5:8C:7A:A3:6D:AC:33:0E:19:BD:5F:67:9E:04:61
+Public key's random art:
++--[ RSA 3072]----+
+|          ..... .|
+|         = ..  =.|
+|        . =  E. +|
+|         . o...  |
+|        S . o.o  |
+|           o *.. |
+|          o o +.+|
+|           .o+.+o|
+|           .oo...|
++-----------------+
+
 -----BEGIN RSA PRIVATE KEY-----
-MIICWwIBAAKBgQDGYd5/O0mBobEVgRx6JSDSF0yVFVYGqhl6UnwF5dMOddy7/kXY
-e0z6X5APINoaMqXH40griuSPt3pZ3cAo+A5hLRVINl7drXF/2tFjM62FZO/u1iLj
-8tOo9eo3mUoTvV2CsH/+A6a9X/VgPvvoUjFUYJIBPSEfCQULeyWaklLAPwIDAQAB
-AoGARIwKqmHc+0rYenq7UUVE+vMMBjNyHyllVkvsCMmpzMRS+i5ZCf1I0vZ0O5X5
-ZrX7bH8PL+R1J2eZgjXKMR3NMZBuyKHewItD9t2rIC0eD/ITlwq3VybbaMsw666e
-INxSmax+dS5CEcLevHHP3c+Q7S7QAFiWV43TdFUGXWJktIkCQQDPQ5WAZ+/Tvv0Q
-vtRjXMeTVaw/bSuKNUeDzFkmGyePnFeCReNFtJLE9PFSQWcPuYcbZgU59JTfA5ac
-Un+cHm31AkEA9Qek+q7PcJ+kON9E6SNodCZn6gLyHjnWrq4tf8pZO3NvoX2QiuD4
-rwF7KWjr6q1JzADpLtwXnuYEhyiLFjJA4wJAcElMCEnG2y+ASH8p7z7HfKGQdLg/
-O1wMB3JA5e0WLK5lllUogI4IaZ3N02NNY25+rLBDqpc/w+ZcxQnIypqNtQJATs9p
-ofON5wSB1oUBbhckZo9fxuWxqEUkJsUA/2Q+9R843XE8h166vdc1HOmRT8bywHne
-hmLl+gazmCFTMw1wzwJAHng+3zGUl4D8Ov3MPFD6hwYYK6/pEdtz/NUsCSazF7eK
-XuuP+DXPHNhXOuF1A3tP74pfc/fC1uCUH2G5z3Fy0Q==
+MIIG5QIBAAKCAYEA8nNWVoZwi7fYKSYxm+lidTbMQEHh9pLXV8P1N2FDwqhNc6Z7
+1VqQw6gIDyzxBjZf3ldyPb6xAcxikvk2XCxVGqDaP7E1rfnCDSqxcgLGr1/TlRH3
+fFMFPqzMgs0GbDoKi7WuhcNMH1yIRk4uPVSUY3IclcE9sNvMpNVY77c9tYeLyoLg
+G8A2ljlSjbHXDmR8E1C+WcOvjFKQunpv29zIOvfp3Wuw0g3C6E26RyvYOnNNq6bm
+DHXdyIkxYNPwxyfBDGq/WhnrFqQTyEqulVLVVIpK+3fIdtmH/4OpOwxzKM0q2k9C
+zL+AI1JqlFoePeIKYQYAOssUBwB0VBDbhyo8f7txBs6OB1phPcmbkAo2bJfL01UT
+315omqlGWjvvDRjDEeRAfXnWu/UVLIs1MNn09l2+fKwgZyOZIQEk5gGgb+xO7pZp
+5W1IuLxB1DMatQjYK1L1MG8KUebovTyumAj8kOaCJaqa+EWcPvsH2wJbzp7JZV7x
+AMnNQGLHHccY9Iv9AgMBAAECggGAY6qwOl+H0pff34KY8Cvv0pmVO3EZ7+ALcEPZ
+AUwVq+a9+CXfUCSr0BK6RWrHJjOiE2fFScajQ0b+2sZh52A+F/U54ub5VyyN7hW3
+3lNv09ZDx0scftouFMa6R6b61fDCNw6X0Nso5teGGci3z3R+j6bwQ0uuDY89wCEI
+L3sXfhFbkQNeF2pbxkOIQUFzU7JRIyuVJJeCCeBDKLV+WFaZZUxWsjQ7+w23KwIQ
+6oMGn3iY3FB+z26AnphvWQunNmeLcq11lpwTpI4XB21FMLFVA4FlawLhMP6ca2OA
+rp1HsTPqthxtFC//KXHinN1AYjQERD2NcWvR6o1GYzcSJo+UHPVq0WZ/qdbdWSd5
+zuqts0diufYcNgrA/vP24cKGEpf58n5itDM+rUMFlR0U/r52xkQ4gKQlCFtcdk6p
+vap/8QWLSYGgQEsamNHJz5jhunES/B9TqldcZTrgciTVjychnPy21vfIB1zCQl3L
+ur8N8hObB2tzrhvwtxpq/SDPsnIBAoHBAP4UzzGg2OD1STUiLBQTqlXWJpLC+mMC
+ct3RBdEPsKSu5A6ZoDXjhCQGqidNzm8zbYqWPlS9UZLpP5vybp19Hi/hEbzbNMgv
+qwfuSl2WBVQcSPA8g5k6n13+MUgMswyr41wt5wZDkVxiIXYKTwDso5WFsP47YYsH
+xyG7tO8N0WcB3b2b6FbryiKDweiJO/oXI+ne/WIEcx8qiWbX0RaZf5ig4MKbESIj
+uVXLoEvrgaSogl3gCssZ/Rhd4hfMEysZgQKBwQD0SAsWOOjZlH44KXHUKvp8zdi1
+0tvRpj8fPzkv8q89rozJ/QEyg89I9PUllT8Kg5AhdWBSAZpeldLtTG6M+ZOaQCS1
+kevmtJZqalEbmxj9JMjDaFLjVfUj4JdNS0vNgwS54XC2TEKmoAwuDfChwAQIVw/a
+Dg7uMnwg5gPuNUEiAbDzbu4N15Ed7ZjU7+YD8TmW3aAfbDo92dKs8BNzMc+x+xPT
+VVUbbL05SeB6t+/sPVzdfb+gEWmPZvxFKbMFGH0CgcEA25yU8ep7SwX4PurIepn8
+0rCFKrlIvqdxLXTruCX9onpgOJZkue9rWdBA5jQESG2FMxJl0czJY1gkWOCU558r
+pmq9cCVwW8+H3xUCRh3TgT3pdWx4usrG0PFTG1mOsVco75XlZ66jR9yx21tRqhTf
+1+90ampdB1Szs96Qp0ekRYqBZNALZIXtcyO7SQjFRIcXzZ+mF0CNWNTlanUEDqqi
+dQQpAUwS5RVHrnaQmM54gh/8frOV4LMX5iXPtI35y3kBAoHBAK3RqZnpar6Pq/R4
+gReQcWZm7qIKZQHbwqSQTzfv/oPsdR+xGK4CHnfyXZGdjhsEHIXCSIRxwY82Gr1h
+mnRTjpS+uaA9MdeafkFAX3A+P1y5VOW+kORUoLcjs2V7g3e7l5xQbuw4fWClF6WZ
+o7TktbndgVkOkc8qOysJoJhaansxKuE8+MT+s7LRAuPZxBjgB7NzBeH2F8c0/HSz
+65MXSYuCdcv1tzZpfIeMKuAHi//UdgTg2n6ihTNV+4GXN1wT3QKBwQCbC2bX/t1p
+0ppb1U428MhnTmj/1SGhklAnl0dugj2zxYfSwBUJwZ7cLXRazCvKggpdO4KTMc4S
+vVt9VIfAig7fhtZvYspKQV9sjsusbjfAuCYgw1k3zh/n66UIirCW8+A9/GbEa2YU
+kXYsIuRqViBBUWp38b9MW+j6r0EJqHdaamZtEoNUwej/tzrbn0bizthgqAMjeEFf
+++oWYiAkjYtJZOTLyoPoxGCX+7/n3CwQugNndupNAGwG+NKD+yb+3OE=
 -----END RSA PRIVATE KEY-----
diff --git a/tests/tls-wrong-permittedPeer.sh b/tests/tls-wrong-permittedPeer.sh
index 709e35f..f6287e7 100755
--- a/tests/tls-wrong-permittedPeer.sh
+++ b/tests/tls-wrong-permittedPeer.sh
@@ -6,5 +6,5 @@ echo 'Send Message...'
 ./send -t 127.0.0.1 -p $TESTPORT -m "testmessage" -T -a "name" -x ${srcdir}/tls-certs/ca.pem -y ${srcdir}/tls-certs/cert.pem -z ${srcdir}/tls-certs/key.pem -P "wrong name" -v 2>&1 | tee librelp.out.log
 
 stop_receiver
-check_output "librelp: auth error: authdata:'DNSname: rsyslog; ', ecode 10034, emsg 'no permited name found'"
+check_output "librelp: auth error: authdata:'DNSname: testbench.rsyslog.com; DNSname: rsyslog-client; ', ecode 10034, emsg 'no permited name found'"
 terminate
